src/udisks2block.cpp

```diff
@@ -533,41 +518,53 @@ void UDisks2::Block::rescan(const QString &dbusObjectPath)
     QVariantMap options;
     arguments << options;
 
-    NemoDBus::Interface *blockDeviceInterface = new NemoDBus::Interface(
-            this, m_connection, UDISKS2_SERVICE, dbusObjectPath, UDISKS2_BLOCK_INTERFACE);
-    NemoDBus::Response *response = blockDeviceInterface->call(UDISKS2_BLOCK_RESCAN, arguments);
-    response->onError([this, dbusObjectPath, blockDeviceInterface](const QDBusError &error) {
-        qCDebug(lcMemoryCardLog) << "UDisks failed to rescan object path" << dbusObjectPath << ", error type:" << error.type() << ",name:" << error.name() << ", message:" << error.message();
-    });
-    connect(response, &QObject::destroyed, [blockDeviceInterface] {
-        delete blockDeviceInterface;
+    QDBusInterface blockDeviceInterface(UDISKS2_SERVICE,
```

Comment:
Constructing `QDBusInterface` comes with cost of doing a blocking introspection. That's why its uses were removed. Alternatives to that would be creating appropriate `QDBusAbstractInterface` subclass or just using `QDBusConnection::send` methods.

After reading the comments in the bug, I understand why you did this. Odd that it wasn't caught before. But I guess fixing this should be part of the original bug and not done here, so I'm just not commenting the revert any further.

Reply:
I just reverted the commit to have the PR working, fixing this should be done in the original bug yes.